js/src/common/utils/mapRoutes.ts

```diff
@@ -6,20 +8,23 @@ import DefaultResolver from '../resolvers/DefaultResolver';
  * to provide each route with the current route name.
  *
  * @see https://mithril.js.org/route.html#signature
- * @param {Object} routes
- * @param {String} [basePath]
- * @return {Object}
  */
-export default function mapRoutes(routes, basePath = '') {
-  const map = {};
+export default function mapRoutes(
+  routes: Record<string, RouteItem<Record<string, unknown>, Component<{ routeName: string; [key: string]: unknown }>, Record<string, unknown>>>,
```

Comment:
Here's another batch where extracting + composing into simpler types would be easier to parse. And ideally, reuse between here and Application